### PR TITLE
[1/N] Apply py39 ruff fixes

### PR DESCRIPTION
**Pull request by cyyever**
cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Comment by Skylion007** [comment]
pyproject.toml needs to be extensively updated along with a few other changes.

**Comment by cyyever** [comment]
> pyproject.toml needs to be extensively updated to for both the formatter and ruff minimum Python version

They are left as last updates to enable incremental PRs.

**Comment by cyyever** [comment]
@Skylion007 May be worth considering it.

File: mypy-strict.ini [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # files.
 
 [mypy]
-python_version = 3.8
+python_version = 3.9
```
Comment (Skylion007) [inline mypy-strict.ini:9]: NIt: Should be it's own PR for easy reversion

**Review by Skylion007 — APPROVED** [review]

File: mypy-strict.ini (hunk 1 above)
Comment (cyyever) [reply mypy-strict.ini:9]: Because refactor is from ruff
